From 70ed1e0c018131737b9f332ff903963de0963adf Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 15:03:54 +0200
Subject: [PORT FROM MAIN] Telephony: Fix for com.android.phone has stopped during MTBF

BZ: 16858

This patch handles the exception ConcurrentModificationException
found during MTBF test.

Change-Id: Ica2e7a52c03c9f2c561ed06807afdce439565e68
Orig-Change-Id: I9647601ddac08bb55b7dc5f9837662bcf76d6937
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/75449
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../android/internal/telephony/CallManager.java    |   54 +++++++++++---------
 1 files changed, 29 insertions(+), 25 deletions(-)

diff --git a/src/java/com/android/internal/telephony/CallManager.java b/src/java/com/android/internal/telephony/CallManager.java
index f32fcf0..b818ffa 100644
--- a/src/java/com/android/internal/telephony/CallManager.java
+++ b/src/java/com/android/internal/telephony/CallManager.java
@@ -31,6 +31,7 @@ import android.telephony.Rlog;
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.ConcurrentModificationException;
 import java.util.List;
 
 
@@ -1850,32 +1851,35 @@ public final class CallManager {
     public String toString() {
         Call call;
         StringBuilder b = new StringBuilder();
-
-        b.append("CallManager {");
-        b.append("\nstate = " + getState());
-        call = getActiveFgCall();
-        b.append("\n- Foreground: " + getActiveFgCallState());
-        b.append(" from " + call.getPhone());
-        b.append("\n  Conn: ").append(getFgCallConnections());
-        call = getFirstActiveBgCall();
-        b.append("\n- Background: " + call.getState());
-        b.append(" from " + call.getPhone());
-        b.append("\n  Conn: ").append(getBgCallConnections());
-        call = getFirstActiveRingingCall();
-        b.append("\n- Ringing: " +call.getState());
-        b.append(" from " + call.getPhone());
-
-        for (Phone phone : getAllPhones()) {
-            if (phone != null) {
-                b.append("\nPhone: " + phone + ", name = " + phone.getPhoneName()
-                        + ", state = " + phone.getState());
-                call = phone.getForegroundCall();
-                b.append("\n- Foreground: ").append(call);
-                call = phone.getBackgroundCall();
-                b.append(" Background: ").append(call);
-                call = phone.getRingingCall();
-                b.append(" Ringing: ").append(call);
+        try {
+            b.append("CallManager {");
+            b.append("\nstate = " + getState());
+            call = getActiveFgCall();
+            b.append("\n- Foreground: " + getActiveFgCallState());
+            b.append(" from " + call.getPhone());
+            b.append("\n  Conn: ").append(getFgCallConnections());
+            call = getFirstActiveBgCall();
+            b.append("\n- Background: " + call.getState());
+            b.append(" from " + call.getPhone());
+            b.append("\n  Conn: ").append(getBgCallConnections());
+            call = getFirstActiveRingingCall();
+            b.append("\n- Ringing: " +call.getState());
+            b.append(" from " + call.getPhone());
+
+            for (Phone phone : getAllPhones()) {
+                if (phone != null) {
+                    b.append("\nPhone: " + phone + ", name = " + phone.getPhoneName()
+                            + ", state = " + phone.getState());
+                    call = phone.getForegroundCall();
+                    b.append("\n- Foreground: ").append(call);
+                    call = phone.getBackgroundCall();
+                    b.append(" Background: ").append(call);
+                    call = phone.getRingingCall();
+                    b.append(" Ringing: ").append(call);
+                }
             }
+        } catch (ConcurrentModificationException e) {
+            Rlog.e(LOG_TAG, "CallManager toString Exception:" + e);
         }
         b.append("\n}");
         return b.toString();
-- 
1.7.4.1

